web/styles/compose.css

```diff
@@ -342,8 +342,32 @@
         var(--color-message-content-container-border-over-limit);
 }
 
+.edit-content-container .message_edit_content.over-limit,
+.edit-content-container .message_edit_content.over-limit:focus {
+    background-color: var(
+        --color-compose-message-content-background-over-limit
+    );
+}
+
+#message-content-container:has(.new_message_textarea.approaching-limit),
+#message-content-container:has(.new_message_textarea.approaching-limit:focus),
+.edit-content-container:has(.message_edit_content.approaching-limit),
+.edit-content-container:has(.message_edit_content.approaching-limit:focus) {
+    box-shadow: 0 0 0 1pt
+        var(--color-message-content-container-border-approaching-limit);
+}
+
+.edit-content-container .message_edit_content.approaching-limit,
+.edit-content-container .message_edit_content.approaching-limit:focus {
```

Comment:
I think there is no need for `edit-content-container` here since `message_edit_content` is a unique class.